test/toolbox_image.sh

```diff
@@ -288,203 +110,88 @@ run_test()
-            || return 1
-}
-
-test_execute_update_files_invalid_block_device_nok()
-{
-    target_device="/dev/loop100"
-    chroot "${rootfs_dir}" "${UPDATE_FILES_COMMAND}" -s "${UPDATE_ROOTFS_SOURCE}" -d "${target_device}" || return 0
-}
-
-test_execute_update_files_target_device_ok()
-{
-    chroot "${rootfs_dir}" "${UPDATE_FILES_COMMAND}" -s "${UPDATE_ROOTFS_SOURCE}" -d "${LOOP_STORAGE_DEVICE}" || return 1
-
-    update_target="$(mktemp -d)"
-    mount -t auto -v "${LOOP_STORAGE_DEVICE}p2" "${update_target}" || return 1
-
-    rsync --exclude-from "${UPDATE_EXCLUDE_LIST_FILE}" -c -a -x --dry-run \
-        "${rootfs_dir}${UPDATE_ROOTFS_SOURCE}/" "${update_target}/" || return 1
-
-    umount "${update_target}"
-
-    if [ -z "${update_target##*/tmp/*}" ]; then
-        rm -rf "${update_target}"
-    fi
-}
-
-test_execute_start_update_rsync_ignore_file_not_found_nok()
-{
-    cp "${TEST_UPDATE_ROOTFS_FILE}" "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    rm "${rootfs_dir:?}${SYSTEM_UPDATE_DIR:?}/"*_exclude_list.txt
-    "${rootfs_dir}${START_UPDATE_COMMAND}" "${rootfs_dir}" "${update_mount}" "${LOOP_STORAGE_DEVICE}" || return 0
-}
-
-test_execute_start_update_partition_table_not_found_nok()
-{
-    cp "${TEST_UPDATE_ROOTFS_FILE}" "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    rm "${rootfs_dir:?}${SYSTEM_UPDATE_DIR:?}/"*.table
-    "${rootfs_dir}${START_UPDATE_COMMAND}" "${rootfs_dir}" "${update_mount}" "${LOOP_STORAGE_DEVICE}" || return 0
-}
-
-test_execute_start_update_update_rootfs_corrupt_nok()
-{
-    cp "${TEST_UPDATE_ROOTFS_FILE}" "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    echo "Append this data to corrupt the archive" >> "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    "${rootfs_dir}${START_UPDATE_COMMAND}" "${rootfs_dir}" "${update_mount}" "${LOOP_STORAGE_DEVICE}" || return 0
-}
-
-test_execute_start_update_multiple_update_rootfs_files_nok()
-{
-    cp "${TEST_UPDATE_ROOTFS_FILE}" "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    cp "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}" \
-        "${update_mount}/rootfs-v2.1.0.tar.xz"
-    "${rootfs_dir}${START_UPDATE_COMMAND}" "${rootfs_dir}" "${update_mount}" "${LOOP_STORAGE_DEVICE}" || return 0
-}
-
-test_execute_start_update_successful_update_ok()
+test_jedi_exclude_list_exists()
 {
-    cp "${TEST_UPDATE_ROOTFS_FILE}" "${update_mount}/${TEMP_TEST_UPDATE_ROOTFS_FILE}"
-    "${rootfs_dir}${START_UPDATE_COMMAND}" "${rootfs_dir}" "${update_mount}" "${LOOP_STORAGE_DEVICE}" || return 1
-
-    update_target="$(mktemp -d)"
-    mount -t auto -v "${LOOP_STORAGE_DEVICE}p2" "${update_target}"
-
-    rsync --exclude-from "${UPDATE_EXCLUDE_LIST_FILE}" -c -a -x --dry-run \
-        "${rootfs_dir}${UPDATE_ROOTFS_SOURCE}/" "${update_target}/" || return 1
-
-    umount "${update_target}"
-
-    if [ -z "${update_target##*/tmp/*}" ]; then
-        rm -rf "${update_target}"
-    fi
+    test -f "${toolbox_root_dir}${JEDI_EXCLUDE_LIST_FILE}" || return 1
```

Comment:
test if the file is readable, just the mere existence is not enough

Reply:
as discussed, the script is executed in a 'root' environment and the file is always readable by root. The intention of the test is to test if the file is there, other tests will fail if the content of the file is not correct.